### PR TITLE
logger fixes

### PR DESCRIPTION
**Pull request by AlekId**

### CODE REVIEW

File: SynLog.pas [hunk 1]
```diff
@@ -2243,8 +2243,10 @@ class procedure TSynMapFile.Log(W: TTextWriter; aAddressAbsolute: PtrUInt;
     exit;
   {$ifdef FPC}
   s := BacktraceStrFunc(pointer(aAddressAbsolute));
-  if Pos('SynLog.pas',s)=0 then // don't log internal calls
+  if Pos('SynLog.pas',s)=0 then begin // don't log internal calls
     W.AddShort(s);
+    W.AddShort(#14);
```
Comment (synopse) [inline SynLog.pas]: what is this #14 for?

File: SynLog.pas [hunk 2]
```diff
@@ -5342,7 +5346,7 @@ procedure TSynLogFile.ProcessOneLine(LineBeg, LineEnd: PUTF8Char);
   if (fCount<=fLineHeaderCountToIgnore) or (LineEnd-LineBeg<24) then
     exit;
   if fLineLevelOffset=0 then begin
-    if (fCount>50) or not (LineBeg[0] in ['0'..'9']) then
+    if  not (LineBeg[0] in ['0'..'9']) then
```
Comment (synopse) [inline SynLog.pas]: why is fCount>50 wrong?

File: SynLog.pas [hunk 3]
```diff
@@ -5682,6 +5686,7 @@ procedure TSynLogFileView.AddInMemoryLine(const aNewLine: RawUTF8);
     tm: cardinal;
 begin
   tm := fThreadMax;
+  fLineLevelOffset:=0;
```
Comment (synopse) [inline SynLog.pas]: this will break the expected behavior of this method, for sure

File: SynLog.pas (hunk 2 above)
Comment (AlekId) [reply SynLog.pas]: because it limits viewer to 50 events, that for sure.

File: SynLog.pas (hunk 1 above)
Comment (AlekId) [reply SynLog.pas]: this is lifehack to break stacktrace in viewer -- I use it to break call stack to separate lines. Don''t know but #10 didnt worked.
ttt

File: SynLog.pas (hunk 3 above)
Comment (AlekId) [reply SynLog.pas]: it's buggy, the multithreaded Lazarus app just don't able to send log to viewer. Or viewer can't read it

Comment (AlekId) [reply SynLog.pas]: that way it works, the other **dont** . Fix it better, if you can. 
My goal is to make logger functional for Lazarus app. I **made** it functional. I don't have time to delve into spaghetti, which implements logger logic.

File: SynLog.pas (hunk 2 above)
Comment (synopse) [reply SynLog.pas]: it is expected to have the header available and discovered within the 50 first text files - this won't limit t 50 events; if it is not the case, then the log format is not correct

File: SynLog.pas (hunk 3 above)
Comment (synopse) [reply SynLog.pas]: https://stackoverflow.com/help/mcve is always better than carping I guess

**Comment by synopse** [comment]
No feedback since more than a year.
Closed.

**Pull request closed without merging**